### PR TITLE
Operator retry test fix attempt

### PR DESCRIPTION
**Pull request by akarnokd**
Maybe the error function was not emitting events properly.

### CODE REVIEW

File: src/main/java/rx/observers/TestSubscriber.java [hunk 1]
```diff
@@ -227,9 +227,9 @@ public void awaitTerminalEvent() {
      * @throws RuntimeException
      *          if the Subscriber is interrupted before the Observable is able to complete
      */
-    public void awaitTerminalEvent(long timeout, TimeUnit unit) {
+    public boolean awaitTerminalEvent(long timeout, TimeUnit unit) {
```
Comment (zsxwing) [inline src/main/java/rx/observers/TestSubscriber.java]: This is a breaking change.

Comment (akarnokd) [inline src/main/java/rx/observers/TestSubscriber.java]: Any suggestion how I can sneak out the await's result?

Comment (zsxwing) [inline src/main/java/rx/observers/TestSubscriber.java]: Throw `TimeoutException`? It's a bit minor breaking change. @benjchristensen Is it fine?

Comment (zsxwing) [inline src/main/java/rx/observers/TestSubscriber.java]: Or add a new method?

Comment (akarnokd) [inline src/main/java/rx/observers/TestSubscriber.java]: I've changed it back and made it throw a RTE with TimeoutException as cause.

**Pull request merged**